packages/adblocker/src/filters/network.ts

```diff
@@ -775,16 +775,6 @@ export default class NetworkFilter implements IFilter {
         filterIndexEnd -= 1;
       }
 
-      // Remove leading '*' if the filter is not hostname anchored.
-      if (
-        getBit(mask, NETWORK_FILTER_MASK.isHostnameAnchor) === false &&
-        filterIndexEnd - filterIndexStart > 0 &&
-        line.charCodeAt(filterIndexStart) === 42 /* '*' */
-      ) {
-        mask = clearBit(mask, NETWORK_FILTER_MASK.isLeftAnchor);
-        filterIndexStart += 1;
-      }
```

Comment:
Is this related to the fix of filters ending with `^`?

Reply:
I think this is a kind of optimisation. There's a test relevant to this: `*bar^`. Also see https://ghostery.slack.com/archives/C06JX8NJVMF/p1712045816656759